### PR TITLE
PR to update optimised params and remove bams after bwamem alignment

### PR DESCRIPTION
**Pull request by Ales-ibt**
PR to update optimised kmer for sourmash, coverage threshold on bwamem, and remove bams after bwamem alignment. The $USER var was added to codon config avoid writing permission conflicts on workdir.

### CODE REVIEW

File: bin/bam2cov_filt.py [hunk 1]
```diff
@@ -114,7 +114,7 @@ def FP_control(out_root, genomes_len, unique_matches, ave_read_len):
 
 def main():
     parser = argparse.ArgumentParser(
-        description="This script process the bam file generated using bwa-mem2 with the flag -M. The bam file has to be filtered using -F4 -F256 flags, then sorted and indexed. The output of this script is a table of relative abundance of genomes based on unique mappin greads. Only genomnes having a mean depth coverage >0.1 are considered."
+        description="This script process the bam file generated using bwa-mem2 with the flag -M. The bam file has to be filtered using -F4 -F256 flags, then sorted and indexed. The output of this script is a table of relative abundance of genomes based on unique mappin greads. Only genomnes having a mean depth coverage >0.01 are considered."
```
Comment (chrisAta) [inline bin/bam2cov_filt.py]: small typo `mappin greads` -> `mapping reads`

**Review by chrisAta — APPROVED** [review]
Looks really good to me! Just a tiny typo spotted

File: modules/local/align/bwamem2_pysam.nf [hunk 2]
```diff
@@ -32,22 +31,32 @@ process ALIGN_BWAMEM2 {
         | samtools sort -@ ${task.cpus} -O bam - -o ${prefix}_sorted.bam
     samtools index -@ ${task.cpus} ${prefix}_sorted.bam
 
+    echo " ---> processing bam file"
+    bam2cov_filt.py \\
+        --bwa_bam ${prefix}_sorted.bam \\
+        --prefix ${prefix}_u_relab_01
+
+    echo " ---> removing bam file"
+    rm -rf *.bam *.bai
```
Comment (KateSakharova) [inline modules/local/align/bwamem2_pysam.nf]: only `rm` should be enough... it is just a file, not folder

**Review by KateSakharova — APPROVED** [review]
Looks good!

File: bin/bam2cov_filt.py [hunk 3]
```diff
@@ -72,7 +72,7 @@ def bam_parser(bam_file):
 
 def FP_control(out_root, genomes_len, unique_matches, ave_read_len):
     unique_thres01 = []
-    cov_threhold = 0.1
+    cov_threhold = 0.01
```
Comment (mberacochea) [inline bin/bam2cov_filt.py]:
```suggestion
    COV_THREHOLD = 0.01
```
Using upper case is a convention for constants

File: modules/nf-core/sourmash/sketch/main.nf [hunk 4]
```diff
@@ -18,8 +18,7 @@ process SOURMASH_SKETCH {
     task.ext.when == null || task.ext.when
 
     script:
-    // required defaults for the tool to run, but can be overridden
-    def args = task.ext.args ?: "dna --param-string 'scaled=1000,k=51,abund'"
+    def args = task.ext.args ?: "dna --param-string 'scaled=1000,k=21,abund'"
```
Comment (mberacochea) [inline modules/nf-core/sourmash/sketch/main.nf:21]: Can you please create the diff of the module?
```bash
nf-core module patch sourmash_sketch ( I think that is the module name )
```

**Review by mberacochea — CHANGES_REQUESTED** [review]
The is one module path / diff missing from the nf-core module sourmash / sketch.

Other than that it looks good to me

File: bin/bam2cov_filt.py (hunk 1 above)
Comment (Ales-ibt) [reply bin/bam2cov_filt.py]: Thanks!

File: bin/bam2cov_filt.py (hunk 3 above)
Comment (Ales-ibt) [reply bin/bam2cov_filt.py]: Added.

File: modules/nf-core/sourmash/sketch/main.nf (hunk 4 above)
Comment (Ales-ibt) [reply modules/nf-core/sourmash/sketch/main.nf:21]: Done!

**Review by mberacochea — APPROVED** [review]
Thank you @Ales-ibt. It looks good to me :+1:

**Comment by mberacochea** [comment]
The linters are angry, let me know if you need a hand fixing those

**Comment by Ales-ibt** [comment]
Yes please, I'd like to fix the lining problems and rid of the github-actions message `This PR is against the master branch`

**Pull request merged**